common/engine_conf/message_helper.hh

```diff
@@ -0,0 +1,230 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+#ifndef CCE_CONFIGURATION_MESSAGE_HELPER_HH
+#define CCE_CONFIGURATION_MESSAGE_HELPER_HH
+#include <absl/container/flat_hash_map.h>
+#include <absl/strings/str_split.h>
+#include "common/engine_conf/state.pb.h"
+
+#ifdef LEGACY_CONF
+#error This library should not be compiled.
+#endif
+
+namespace com::centreon::engine::configuration {
+
+/**
+ * @brief Error counter, it contains two attributes, one for warnings and
+ * another for errors.
+ */
+struct error_cnt {
+  uint32_t config_warnings = 0;
+  uint32_t config_errors = 0;
+};
+
+/* Forward declarations */
+class command_helper;
+class connector_helper;
+class contact_helper;
+class contactgroup_helper;
+class host_helper;
+class hostdependency_helper;
+class hostescalation_helper;
+class hostgroup_helper;
+class service_helper;
+class servicedependency_helper;
+class serviceescalation_helper;
+class servicegroup_helper;
+class timeperiod_helper;
+class anomalydetection_helper;
+class severity_helper;
+class tag_helper;
+class state_helper;
+
+using ::google::protobuf::Message;
+
+bool fill_pair_string_group(PairStringSet* grp, const std::string_view& value);
+bool fill_pair_string_group(PairStringSet* grp,
+                            const std::string_view& key,
+                            const std::string_view& value);
+void fill_string_group(StringList* grp, const std::string_view& value);
+void fill_string_group(StringSet* grp, const std::string_view& value);
+bool fill_host_notification_options(uint32_t* options,
+                                    const std::string_view& value);
+bool fill_service_notification_options(uint32_t* options,
+                                       const std::string_view& value);
+
+/**
+ * @brief The base message helper used by every helpers. It defines the common
+ * methods.
+ *
+ */
+class message_helper {
+ public:
+  enum object_type {
+    command = 0,
+    connector = 1,
+    contact = 2,
+    contactgroup = 3,
+    host = 4,
+    hostdependency = 5,
+    hostescalation = 6,
+    hostgroup = 8,
+    service = 9,
+    servicedependency = 10,
+    serviceescalation = 11,
+    servicegroup = 13,
+    timeperiod = 14,
+    anomalydetection = 15,
+    severity = 16,
+    tag = 17,
+    state = 18,
+  };
+
+ private:
+  const object_type _otype;
+  Message* _obj;
+  const absl::flat_hash_map<std::string, std::string> _correspondence;
+  std::vector<bool> _modified_field;
+  bool _resolved = false;
+
+ public:
+  message_helper(object_type otype,
```

Comment:
a comment (what is field_size, correspondence)?

Reply:
done